test/test_utils.py

```diff
@@ -642,5 +645,57 @@ def test_assert_true(self):
             torch.Assert(False, "bar")
 
 
+class TestStandaloneCPPJIT(TestCase):
+    def test_load_standalone(self):
+        build_dir = tempfile.mkdtemp()
+        try:
+            src_path = os.path.join(build_dir, "main.cpp")
+            src = textwrap.dedent("""\
+                #include <iostream>
+                #include <torch/torch.h>
+
+                int main() {
+                    auto x = torch::eye(3);
+                    std::cout << x << std::endl;
+                }
+            """)
+            with open(src_path, "wt") as f:
+                f.write(src)
+
+            exec_path = torch.utils.cpp_extension.load(
+                "standalone_load_test",
+                src_path,
+                build_directory=build_dir,
+                is_python_module=False,
+                is_standalone=True,
+            )
+
+            ext = "exe" if IS_WINDOWS else "o"
+            self.assertEqual(
+                exec_path,
+                os.path.join(build_dir, f"standalone_load_test.{ext}")
+            )
+
+            for shell in [True, False]:
+                r = subprocess.run(
+                    [exec_path],
+                    shell=shell,
+                    stdout=subprocess.PIPE,
+                )
+                self.assertEqual(r.returncode, 0)
```

Comment:
The current Windows exit code `3221225781 = 0xc0000135 = STATUS_DLL_NOT_FOUND`. 
To workground this, You 'll need to patch the `PATH` variable. Take a look at https://github.com/pytorch/pytorch/blob/master/torch/__init__.py#L46-L77 and https://github.com/pytorch/pytorch/blob/master/torch/__init__.py#L122.
```
if IS_WINDOWS:
    run_env = os.environ.copy()
    run_env["PATH"] = additional_paths + run_env["PATH"]

...
subprocess.run(...., env=run_env)
```

Reply:
I think the question is what the API should be. It's moot for my use since Valgrind is Linux only, but someone might find it standalone C++ JIT useful for other applications. Maybe return two strings: `PATH additions` and `exec path`, where the former is just empty on Linux? Or disallow altogether on Windows? (That's what I initially did, but I figured I'd use CI to make sure that it wouldn't work out of the box.) I don't have strong feelings one way or another.